### ISSUE TITLE
Toolbar 'Show more items' button is focused after clicking a button from it

### ISSUE THREAD

**Issue by Dumluregn** [ckeditor/ckeditor5#12178]
## 📝 Provide detailed reproduction steps (if any)

1.  Open full features demo http://localhost:8125/ckeditor5/tests/manual/all-features.html
2.  Reduce the width of the editor so that some buttons are grouped.
3.  Click a button that should move the focus to the editable (e.g. choose language) from the collapsed part of toolbar.

### ✔️ Expected result

Focus is moved to the editable.

### ❌ Actual result

Focus is moved to the "Show more items" toolbar button.

### ❓ Possible solution

It is a regression introduced in #12162. Due to moving the focus when any dropdown is closed, it also affects the toolbar dropdown which is closed after the button's command is executed, so it focuses the toolbar back.

The problem occurs on every type of button - generic one, dropdown and splitbutton.

## 📃 Other details

*   Browser: any
*   OS: any
*   First affected CKEditor version: #12162 - [ck/12125-dropdown-focus](https://github.com/ckeditor/ckeditor5/tree/ck/12125-dropdown-focus) branch
*   Installed CKEditor plugins: toolbar

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Dumluregn** [comment]
A cheap and dirty idea could be checking the `document.activeElement` when dropdown is being closed and don’t focus the button if editable is focused, but I’m not sure if it covers all the cases.

**Comment by Reinmar** [comment]
Could you make a draft PR with this?

**Comment by Dumluregn** [comment]
Yes, I'm on it.

**Comment by Dumluregn** [comment]
PoC: `ck/12178-grouped-toolbar-focus`. cc @Reinmar @mlewand

**Comment by Reinmar** [comment]
cc @ckeditor/qa-team – could you verify this idea?

**Comment by Mgsy** [comment]
@FilipTokarski @Acrophost could you please take a look at it :point\_up:? Tests should cover:

*   Desktop browsers,
*   Mobile devices,
*   All editor types,
*   Different toolbars (the main toolbar, the floating toolbar, the contextual toolbar)

**Comment by mlewand** [comment]
@Mgsy @FilipTokarski @Acrophost please check whether this issue also solves #12186 - because there's a good chance it does. (in this case please update PR so that it closes both issues)

**Comment by FilipTokarski** [comment]
The selection is fine but it's not possible to close the expanded toolbar when using lists split button (I'm not sure whether this PoC should address this as well but reporting anyway. It closes automatically when using other buttons or choosing list style from the plit dropdown):

https://user-images.githubusercontent.com/34380544/182860615-cddac279-b43c-45e9-8bf5-32bcb82da0c0.mov

**Comment by Acrophost** [comment]
It's also not possible to close the expanded toolbar when inserting code block with button - like in case above focus is correct, but it's still worth reporting :)


https://user-images.githubusercontent.com/66953380/182870754-72c0a1b1-b9d9-405a-a7ef-5466e349c2ee.mov

**Comment by Acrophost** [comment]
Same case with inserting media - although here the problem doesn't appear in case of balloon toolbar


https://user-images.githubusercontent.com/66953380/182870954-106fb003-52fb-4aba-899e-f845c10e5469.mov

**Comment by Dumluregn** [comment]
Most of the issues above seem to be the same as #12207, only the reproduction steps are a bit different (using toolbar buttons) - @FilipTokarski @Acrophost please verify it and hide the proper comments if that's the case.

**Comment by FilipTokarski** [comment]
Thanks, you're right 👍 I'll leave the one with inserting image, as it looks a bit different because the focus is still in the toolbar.

**Comment by Dumluregn** [comment]
Yes, the image button doesn't focus the editable - https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-image/src/imageupload/imageuploadui.js#L57-L63. I think it will be worth an actual issue.

**Comment by mlewand** [comment]
Shouldn't we extract [https://github.com/ckeditor/ckeditor5/issues/12178#issuecomment-1205261084](https://github.com/ckeditor/ckeditor5/issues/12178#issuecomment-1205261084) to a separate issue? As [Darek explained](https://github.com/ckeditor/ckeditor5/issues/12178#issuecomment-1205359112) there's no explicit focus change in the current implementation.

**Comment by Acrophost** [comment]
I've tested the fix on different browsers, devices and editor types - for now I don't see any more issues with focus aside from ones that are already reported.

**Comment by Dumluregn** [comment]
Thanks QA, I'll create a PR with the fix.

**Comment by mlewand** [comment]
The [https://github.com/ckeditor/ckeditor5/issues/12178#issuecomment-1205363543](https://github.com/ckeditor/ckeditor5/issues/12178#issuecomment-1205363543) should be fixed by #12229 (on the latest master).
